### PR TITLE
Add max_total_transactions_per_worker for re-creating workers

### PR DESCRIPTION
**Pull request by DarrenTsung**
After they've lived for too long. Looking to narrow down investigation
on insidious bug for requests taking too long after a certain amount
of time.

(optional: @jwilm)

### CODE REVIEW

File: src/executor/mod.rs [hunk 1]
```diff
@@ -163,7 +169,13 @@ impl<D: Deliverable> Future for Executor<D> {
...
                                     counter
                                 );
-                            },
+
+                                self.transactions_processed += 1;
+                                if self.transactions_processed > self.max_total_transactions_per_worker {
+                                    state_changed = true;
+                                    break ExecutorState::Draining;
+                                }
+                            }
```
Comment (coder543) [inline src/executor/mod.rs]: Would it be reasonable to prevent queuing up transactions in excess of the remaining amount that will be processed?

Comment (coder543) [reply src/executor/mod.rs]: discussed offline, lgtm

Comment (DarrenTsung) [reply src/executor/mod.rs]: It should be okay as we want this client to stop processing transactions after the limit, dropped transactions will be retried with another client.

Comment (jwilm) [reply src/executor/mod.rs]: How does this work with in-flight transactions?

Comment (DarrenTsung) [reply src/executor/mod.rs]: In-flight transactions are still tracked and the executor is held in the Draining state until they are finished. However, the receiver is dropped the moment it transitions to the Draining state, so from the pools point of view the executor is gone and will be re-created.

**Pull request merged**